lib/ansible/module_utils/keycloak.py

```diff
@@ -203,3 +219,108 @@ def delete_client(self, id, realm="master"):
         except Exception as e:
             self.module.fail_json(msg='Could not delete client %s in realm %s: %s'
                                       % (id, realm, str(e)))
+
+    def get_client_templates(self, realm='master'):
+        """ Obtains client template representations for client templates in a realm
+
+        :param realm: realm to be queried
+        :return: list of dicts of client representations
+        """
+        url = URL_CLIENTTEMPLATES.format(url=self.baseurl, realm=realm)
+
+        try:
+            return json.load(open_url(url, method='GET', headers=self.restheaders,
+                                      validate_certs=self.validate_certs))
+        except Exception as e:
```

Comment:
Can we enumerate the exceptions here?  Such as at least `ValueError` for failure to parse as JSON?

Reply:
I added ValueError when doing json.load throughout the file. The end-result is still rather similar (a fail_json with slightly changed error message and the stringified exception in the end).

One could add catching of specific url exceptions as well (such as CertificateError, ProxyError, etc.), though the handling would be no different since there is no way to recover from them.